public/app/types/unified-alerting-dto.ts

```diff
@@ -85,7 +84,49 @@ export interface RulerAlertingRuleDTO extends RulerRuleBaseDTO {
   annotations?: Annotations;
 }
 
-export type RulerRuleDTO = RulerAlertingRuleDTO | RulerRecordingRuleDTO;
+export enum GrafanaAlertState {
+  Alerting = 'Alerting',
+  NoData = 'NoData',
+  KeepLastState = 'KeepLastState',
+  OK = 'OK',
+}
+
+export interface GrafanaQueryModel {
+  datasource: string;
+  datasourceUid: string;
+  refId: string;
+  [key: string]: any;
+}
+
+export interface GrafanaQuery {
```

Comment:
Isn't this similar to `DataQuery` ?

Reply:
apparently not , they only share `refId` and `queryType` properties, but others are different. I based this on what the new api accepts

Reply:
I don't know if we should extend it or something then, there'll be type mismatches as the queries in QueryGroupOptions expects a DataQuery | ExpressionQuery

Reply:
I'm not sure either... let's solve it later, when we integrate?